instrumentation-genai/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/utils.py

```diff
@@ -192,38 +192,67 @@ def get_llm_request_attributes(
-            "frequency_penalty"
-        ),
-        GenAIAttributes.GEN_AI_OPENAI_REQUEST_SEED: kwargs.get("seed"),
     }
 
-    if (response_format := kwargs.get("response_format")) is not None:
-        # response_format may be string or object with a string in the `type` key
-        if isinstance(response_format, Mapping):
-            if (
-                response_format_type := response_format.get("type")
-            ) is not None:
+    # Add chat-specific attributes only for chat operations
+    if operation_name == GenAIAttributes.GenAiOperationNameValues.CHAT.value:
+        attributes.update(
+            {
+                GenAIAttributes.GEN_AI_REQUEST_TEMPERATURE: kwargs.get(
+                    "temperature"
+                ),
+                GenAIAttributes.GEN_AI_REQUEST_TOP_P: kwargs.get("p")
+                or kwargs.get("top_p"),
+                GenAIAttributes.GEN_AI_REQUEST_MAX_TOKENS: kwargs.get(
+                    "max_tokens"
+                ),
+                GenAIAttributes.GEN_AI_REQUEST_PRESENCE_PENALTY: kwargs.get(
+                    "presence_penalty"
+                ),
+                GenAIAttributes.GEN_AI_REQUEST_FREQUENCY_PENALTY: kwargs.get(
+                    "frequency_penalty"
+                ),
+                GenAIAttributes.GEN_AI_OPENAI_REQUEST_SEED: kwargs.get("seed"),
+            }
+        )
+
+        if (response_format := kwargs.get("response_format")) is not None:
+            # response_format may be string or object with a string in the `type` key
+            if isinstance(response_format, Mapping):
+                if (
+                    response_format_type := response_format.get("type")
+                ) is not None:
+                    attributes[
+                        GenAIAttributes.GEN_AI_OPENAI_REQUEST_RESPONSE_FORMAT
+                    ] = response_format_type
+            else:
                 attributes[
                     GenAIAttributes.GEN_AI_OPENAI_REQUEST_RESPONSE_FORMAT
-                ] = response_format_type
-        else:
-            attributes[
-                GenAIAttributes.GEN_AI_OPENAI_REQUEST_RESPONSE_FORMAT
-            ] = response_format
+                ] = response_format
+
+        service_tier = kwargs.get("service_tier")
+        attributes[GenAIAttributes.GEN_AI_OPENAI_RESPONSE_SERVICE_TIER] = (
+            service_tier if service_tier != "auto" else None
+        )
+
+    # Add embeddings-specific attributes
+    elif (
+        operation_name
+        == GenAIAttributes.GenAiOperationNameValues.EMBEDDINGS.value
+    ):
+        # Add embedding dimensions if specified
+        if "dimensions" in kwargs and kwargs["dimensions"] is not None:
+            attributes["gen_ai.embeddings.dimension.count"] = kwargs[
+                "dimensions"
+            ]
+
+        # Add encoding format if specified
+        if "encoding_format" in kwargs:
+            attributes["gen_ai.request.encoding_formats"] = kwargs[
```

Comment:
This should be a list of encoding_formats, not a single string . On the assert side they would become tuples